api/samples/memtrace_simple.c

```diff
@@ -314,13 +314,31 @@ event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *wher
     DR_ASSERT(instr_is_app(instr_operands));
 
     for (i = 0; i < instr_num_srcs(instr_operands); i++) {
-        if (opnd_is_memory_reference(instr_get_src(instr_operands, i)))
-            instrument_mem(drcontext, bb, where, instr_get_src(instr_operands, i), false);
+        const opnd_t src = instr_get_src(instr_operands, i);
+        if (opnd_is_memory_reference(src)) {
+#ifdef AARCH64
+            /* Memory references involving SVE registers are not supported yet, that work
+             * is coming in i#5844.
```

Comment:
The problem with changing `drutil_insert_get_mem_addr()` to return false if there's an SVE operand is that the samples that use this function `DR_ASSERT` if it's false, so I'd need to update all of them to just skip rather than fail in that case.  The problem with that is that I'm now changing the behaviour of them, because if there's a genuine error they won't fail anymore, they'll skip past the problem operand/instr.

The benefit to these ugly defines and big comments is they're easily found, removed, and don't change any API behaviour.  The 5844 work is nearing completion so these changes won't be around for long.

Reply:
> We had planned to put out a 10.0.0 DynamoRIO release very soon, in the next week. Should we wait?

Yes that would help us.

> The inaccuracies are only with SVE scatter/gather, and not with other SVE operations, right?

Yes, variants with SVE vector register (Z) as part of address operand.

Reply:
The full SVE scatter/gather support is complex: is there a rough ETA?  If it's more than one or two weeks (which seems likely?) should we put out 10.0.0 now, and then in a couple of months put out 10.0.1 or 10.1.0?  So long as we don't break compatibility it is relatively simple to put out a bugfix or minor feature release.  The issue is that the last official release from Feb'22 doesn't work on Ubuntu22 (due to 2 glibc issues) and the drmemtrace view tool from that release doesn't work on the public traces we released last fall.  Multiple users have hit this trying to use that release.

Reply:
> The full SVE scatter/gather support is complex: is there a rough ETA? If it's more than one or two weeks (which seems likely?) should we put out 10.0.0 now, and then in a couple of months put out 10.0.1 or 10.1.0? 

Yes that's best, go ahead with 10.0.0 now.

Reply:
> What if we at least added a user-visible stderr message (under a do-once so we don't get millions) so users are aware of inaccuracies until this is addressed?

@cmannett85-arm has implemented this in `memtrace_simple.c`, `memval_simple.c` and `tracer.cpp`.